core/src/main/java/com/linecorp/armeria/internal/common/RequestContextUtil.java

```diff
@@ -225,20 +225,25 @@ public static void pop(RequestContext current, @Nullable RequestContext toRestor
     public static SafeCloseable invokeHookAndPop(RequestContext current, @Nullable RequestContext toRestore) {
         requireNonNull(current, "current");
 
-        final SafeCloseable closeable = invokeHook(current);
+        final AutoCloseable closeable = invokeHook(current);
```

Comment:
I changed `SafeCloseable` to `AutoCloseable` for better interop with other libraries.
The previous type signature is `hook(Supplier<? extends SafeCloseable> contextHook)`.
If we want to pass a `Scope` class, we have to convert `AutoCloseable` to `SafeCloseable`.
I thought it would be inconvenient.

A returned `SafeCloseable` has no problem for users. `SafeCloseable` maybe not be a good choice for an input parameter.